### ISSUE TITLE
`VisualizeSpec` fails sometimes

### ISSUE THREAD

**Issue by jcranky** [zalando/grafter#48]
And just sometimes, with an error like this:

```
[info] VisualizeSpec
[info] 
[error]  x The example graph must be correctly serialized into .dot format
[error]  'strict digraph {
[error]    "A" [shape=box];
[error]    "B # 1/2" [shape=box];
[error]    "B # 2/2" [shape=box];
[error]    "C # 1/2" [shape=box];
[error]    "C # 2/2" [shape=box];
[error]    "D" [shape=box];
[error]    "B # 1/2" -> "A"
[error]    "B # 2/2" -> "A"
[error]    "C # 1/2" -> "A"
[error]    "C # 1/2" -> "B # 1/2"
[error]    "C # 1/2" -> "B # 2/2"
[error]    "C # 2/2" -> "A"
[error]    "C # 2/2" -> "B # 2/2"
[error]    "D" -> "C # 1/2"
[error]    "D" -> "C # 2/2"
[error]  }' != 'strict digraph {
[error]    "A" [shape=box];
[error]    "B # 1/2" [shape=box];
[error]    "B # 2/2" [shape=box];
[error]    "C # 1/2" [shape=box];
[error]    "C # 2/2" [shape=box];
[error]    "D" [shape=box];
[error]    "B # 1/2" -> "A"
[error]    "B # 2/2" -> "A"
[error]    "C # 1/2" -> "A"
[error]    "C # 1/2" -> "B # 1/2"
[error]    "C # 1/2" -> "B # 2/2"
[error]    "C # 2/2" -> "A"
[error]    "C # 2/2" -> "B # 1/2"
[error]    "C # 2/2" -> "B # 2/2"
[error]    "D" -> "C # 1/2"
[error]    "D" -> "C # 2/2"
[error]  }' (VisualizeSpec.scala:24)
[error] Actual:   ...aph {
[error] ... "A" [shape=box];
[error] ...1/2" [shape=box];
[error] ...2/2" [shape=box];
[error] ...1/2" [shape=box];
[error] ...2/2" [shape=box];
[error] ... "D" [shape=box];
[error] ...> "A"
[error] ...> "A"
[error] ...> "A"
[error] ... 1/2"
[error] ... 2/2"
[error] ...> "A"
[error] ..."B # [2]/2"
[error]   "[D]" ... "[C] # [1]/2"
[error] ..."C # [2]/2"
[error] [}]
[error] Expected: ...aph {
[error] ... "A" [shape=box];
[error] ...1/2" [shape=box];
[error] ...2/2" [shape=box];
[error] ...1/2" [shape=box];
[error] ...2/2" [shape=box];
[error] ... "D" [shape=box];
[error] ...> "A"
[error] ...> "A"
[error] ...> "A"
[error] ... 1/2"
[error] ... 2/2"
[error] ...> "A"
[error] ..."B # [1]/2"
[error]   "[C # 2/2]" ... "[B] # [2]/2"
[error] ..."C # [1]/2"
[error] [  "D" -> "C # 2/2"]
```

**Comment by etorreborre** [comment]
Yes I couldn't figure out this one. As if there was some shared mutable state somewhere.

**Comment by jcranky** [comment]
fixed by #60
